app/domain/authentication/authn_k8s/kubectl_exec.rb

```diff
@@ -181,27 +200,44 @@ def copy(k8s_object_lookup:,
         container: 'authenticator')
         execute(
           k8s_object_lookup: k8s_object_lookup,
-          pod_namespace: pod_namespace,
-          pod_name: pod_name,
-          container: container,
-          cmds: ['tar', 'xvf', '-', '-C', '/'],
-          body: tar_file_as_string(path, content, mode),
-          stdin: true
+          pod_namespace:     pod_namespace,
+          pod_name:          pod_name,
+          container:         container,
+          cmds:              %w(sh),
+          body:              copy_script(path, content, mode),
+          stdin:             true
         )
       end
 
       private
 
-      def tar_file_as_string(path, content, mode)
-        tarfile = StringIO.new("")
-
-        Gem::Package::TarWriter.new(tarfile) do |tar|
-          tar.add_file(path, mode) do |tf|
-            tf.write(content)
-          end
-        end
-
-        tarfile.string
+      def copy_script(path, content, mode)
+        parts = path.split('/')
+        filename = parts.pop
+        dir = parts.join('/')
+
+        # We first copy the content into a temporary file and only then move it to
+        # the desired path as the client polls on its existence and we want it to
+        # exist only when the whole content is present
+        tmp_cert = "#{dir}/tmp_#{filename}"
+
+        # We redirect the output also to a log file on the authn-client container
+        # for debugging purposes.
+        "
+#!/bin/sh
+
+copy_file() {
+  cat > #{tmp_cert} <<EOF
+#{content}
+EOF
+  mv #{tmp_cert} #{path}
+  chmod #{mode} #{path}
```

Comment:
```suggestion
  chmod #{mode} #{path}
  mv #{tmp_cert} #{path}
```
In this case client perspective will be:
File Does Not Exist -> File Exists -> Read File
instead of
File Does Not Exist -> File Exists -> Access Denied
WDYT?

Reply:
nice, thanks